dspace-api/src/main/java/org/dspace/discovery/SolrServiceResourceRestrictionPlugin.java

```diff
@@ -69,7 +69,12 @@ public void additionalSearchParameters(Context context, DiscoverQuery discoveryQ
             if(!authorizeService.isAdmin(context)){
             	StringBuilder resourceQuery = new StringBuilder();
                 //Always add the anonymous group id to the query
-                resourceQuery.append("read:(g0");
+                Group anonymousGroup = groupService.findByName(context,Group.ANONYMOUS);
+                String anonGroupId = "";
+                if(anonymousGroup!=null){
+                    anonGroupId = anonymousGroup.getID().toString();
+                }
+                resourceQuery.append("read:(g"+anonGroupId);
```

Comment:
'resourceQuery.append("read:(g").append(anonGroupId)' might be a tad more efficient, but this works.